techniques/fileDistribution/copyGitFile/2.0/copyFileFromSharedFolder.st

```diff
@@ -80,6 +80,11 @@ bundle agent download_from_shared_folder
       string     => "0",
       ifvarclass => "!enable_suid_${index}.!enable_sgid_${index}";
 
+  defaults:
+      # Add a trailing slash to force considering destination as a directory and avoid replacing it with a file
+      "copyfile[${index}][destination]" string => "${copyfile[${index}][destination]}/", if_match_regex => ".*[^\/]",
+                                    ifvarclass => "pass1.!is_file_${index}";
```

Comment:
I tested it, and it seemed to work well (but it was on Rudder 4.1, I will check with 3.1)